rasa_core/domain.py

```diff
@@ -224,16 +224,24 @@ def input_states(self):
             self.slot_states + \
             self.prev_action_states
 
-    @staticmethod
-    def get_parsing_states(tracker):
+    def get_parsing_states(self, tracker):
         # type: (DialogueStateTracker) -> Dict[Text, float]
 
         state_dict = {}
 
-        # Set all found entities with the state value 1.0
+        # Set all found entities with the state value 1.0, unless they should
+        # be ignored for the current intent and are not associated with a
+        # featurized slot
         for entity in tracker.latest_message.entities:
-            key = "entity_{0}".format(entity["entity"])
-            state_dict[key] = 1.0
+            entity_name = entity.get("entity")
+            should_ignore_entity = (tracker.latest_message.intent.get("name")
+                                    in self.intents_ignore_entities)
+            entity_is_featurized_slot = (entity_name in tracker.slots.keys()
+                                         and tracker.slots[entity_name].type_name
+                                         != 'unfeaturized')
+            if not should_ignore_entity or entity_is_featurized_slot:
```

Comment:
is this the desired behaviour? it ignores all entities if they are not mapped to a slot?

Reply:
The original statement was this: `if (not should_ignore_entity or (should_ignore_entity and entity_is_featurized_slot))`, but that one equates to the same thing. So ignore the entities only if they're part of an intent in the list and not a featurized slot

Reply:
thanks :+1: why do we featurize the entity if it maps to a slot and is ignored? won't the slot be featurized and there is essentially no need to featurize the entity as well?

Reply:
good point, i think it was just to keep it how we do it currently. @JoeyFaulkner @Ghostvv was there a particular reason we decided to do it this way?

Reply:
We need to featurize the entity for the cases like `correct{"price": "expensive"}`, in this case the slot was set before, but we need to react with certain entity to the fact that it was input again

Reply:
but why would you ignore the entity in this case in the first place? or should we rather have something like:
```
intents:
- greet:
   use_entities: ["name"]
```
meaning the intent will ignore all entities but the name?

Reply:
yeah i agree, in that case you shouldn't unfeaturize entities @Ghostvv . but i'm not sure we need your suggestion @tmbo - i don't see a case where you'd want to pay attention to one but not the rest?

Reply:
but isn't the example @Ghostvv posted an example for this - ignore all the entities, but the "price" entity?

Reply:
no because in the `correct` case you want to pay attention to all entities, hence why I said you shouldn't ignore them in the first place

Reply:
ok in that case and if there are no objections I'd suggest to:
- not featurize ignored entities, even if there are slots
- keep the simple boolean flag, we can still later extend our `"use_entities": false` to `"use_entities": ["name"]` later if we need it

Reply:
@Ghostvv no we're not defeaturizing slots, we're just not featurizing the entities even if their slots are featurized. So like this: 
```
            should_use_entity = (tracker.latest_message.intent.get("name") not
                                 in self.intents_ignore_entities)
            if should_use_entity:
                key = "entity_{0}".format(entity["entity"])
                state_dict[key] = 1.0
```

Reply:
but we already discussed it? it will cause different feature vectors during prediction and train times if train stories were without entities and, it will cause the same branching if entities are still present in train story. I agree that resulting behavior is the same, but this way we create somewhat hidden behavior that can be difficult to understand and debug for users

Reply:
i'm not entirely sure you're thinking of the same thing as us? of course there will still be tracker branching, but it's going to do that anways since they're associated with featurized slots

Reply:
ok, I think i get now

Reply:
this is super hard to discuss via github 🙈